disperser/dataapi/subgraph/api.go

```diff
@@ -74,13 +74,26 @@ func (a *api) QueryBatchesByBlockTimestampRange(ctx context.Context, start, end
 		"blockTimestamp_gte": graphql.Int(start),
 		"blockTimestamp_lte": graphql.Int(end),
 	}
+	skip := 0
 	query := new(queryBatchesByBlockTimestampRange)
-	err := a.uiMonitoringGgl.Query(ctx, query, variables)
-	if err != nil {
-		return nil, err
+	result := make([]*Batches, 0)
+	for {
+		variables["first"] = graphql.Int(maxEntriesPerQuery)
+		variables["skip"] = graphql.Int(skip)
+
+		err := a.uiMonitoringGgl.Query(ctx, &query, variables)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(query.Batches) == 0 {
+			break
+		}
+		result = append(result, query.Batches...)
+		skip += maxEntriesPerQuery
```

Comment:
Should we use cursor based pagination instead? what happens if the entries are updated between these calls?